extend_vc_commands.js

```diff
@@ -0,0 +1,194 @@
+function vimActive(currentApplication, mode) {
+  if (currentApplication === "iTerm" || currentApplication === "Terminal" && mode.lastIndexOf('vim', 0) === 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function vimActiveNormalMode(currentApplication, mode) {
+  if (currentApplication === "iTerm" || currentApplication === "Terminal "&& mode === "vim-normal") {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function vimActiveInsertMode(currentApplication, mode) {
+  if (currentApplication === "iTerm" || currentApplication === "Terminal" && mode === "vim-insert") {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+Commands.extend("sage", function() {
+  if (vimActive(this.currentApplication(), Commands.mode)) {
+    this.key("Escape");
+    this.key(":");
+    this.key("W");
+    this.key("A");
+    this.key("Return");
+    this.stop();
+  }
+});
+
+Commands.extend("trail", function(input) {
+  if (vimActive(this.currentApplication(), Commands.mode)) {
+    this.key("Escape");
+    this.key("?");
+    for (var i = 0; i < input.value.length; i++) {
+      this.key(input.value[i]);
+    }
+    this.key("Return");
+    this.stop();
+  }
+});
+
+Commands.extend("crew", function(input) {
+  if (vimActive(this.currentApplication(), Commands.mode)) {
+    this.key("Escape");
+    this.key("/");
+    for (var i = 0; i < input.value.length; i++) {
+      this.key(input.value[i]);
+    }
+    this.key("Return");
+    this.stop();
+  }
+});
+
+Commands.extend("seltrail", function(input) {
+  if (vimActive(this.currentApplication(), Commands.mode)) {
+    this.key("O");
+    this.key("?");
+    for (var i = 0; i < input.value.length; i++) {
+      this.key(input.value[i]);
+    }
+    this.key("Return");
+    this.stop();
+  }
+});
+
+Commands.extend("selcrew", function(input) {
+  if (vimActive(this.currentApplication(), Commands.mode)) {
+    this.key("/");
+    for (var i = 0; i < input.value.length; i++) {
```

Comment:
@JarrodCTaylor - it would be easier here to use `this.string(input)` - especially because the input may have characters in it that are not valid parameters to the `key` command